src/business_logic/fact_state/state.rs

```diff
@@ -200,14 +198,28 @@ mod test {
 
     #[test]
     fn test_from_cached_state_without_updates() {
-        let mut state_reader = InMemoryStateReader::new(HashMap::new(), HashMap::new());
+        let mut state_reader = InMemoryStateReader::new(
+            HashMap::new(),
+            HashMap::new(),
+            HashMap::new(),
+            HashMap::new(),
+        );
 
         let contract_address = Address(32123.into());
-        let contract_state = ContractState::new([8; 32], Felt::new(109), HashMap::new());
+        let class_hash = [9; 32];
+        let nonce = Felt::new(42);
+        let storage_entry = (contract_address.clone(), [17; 32]);
+        let storage_value = Felt::new(402);
 
         state_reader
-            .contract_states
-            .insert(contract_address, contract_state);
+            .address_to_class_hash
+            .insert(contract_address.clone(), class_hash);
+        state_reader
+            .address_to_nonce
+            .insert(contract_address, nonce);
+        state_reader
```

Comment:
Since this storage insert wasn´t in the previous version of the test, can we delete them?

Reply:
Done